sample/ClassComponents.fs

```diff
@@ -0,0 +1,74 @@
+﻿module ClassComponents
+
+open Lit
+open Fable.Core
+open Fetch
+
+type MouseController =
+    inherit ReactiveController
+    abstract x: float
+    abstract y: float
+type Post = {| usedId: int; id: int; title: string; body: string |}
+
+[<Import("MouseController", from="./controllers.js"); Emit("new $0($1)")>]
+let private createMouseController (host: LitElement): MouseController = jsNative
+
+type ApiController(host: LitElement) as this =
+    do
+        host.addController(this)
```

Comment:
I don't like to use `as this` very much in constructors, because it often raises warnings from the F# compiler and the code generated is quite ugly. If we declare `ReactiveController` as an abstract class we can do this automatically and also add default implementations for the abstract members so users don't need to implement all of them:

```fsharp
[<AbstractClass; AttachMembers>]
type ReactiveController(host: ReactiveControllerHost) =
    do host.addController(jsThis)
    member _.host = host

    abstract hostConnected: unit -> unit
    default _.hostConnected() = ()

    abstract hostDisconnected: unit -> unit
    default _.hostDisconnected() = ()

    abstract hostUpdate: unit -> unit
    default _.hostUpdate() = ()

    abstract hostUpdated: unit -> unit
    default _.hostUpdated() = ()
```

Then your sample becomes:

```fsharp
type ApiController(host: ReactiveControllerHost) =
    inherit ReactiveController(host)
    override this.hostConnected() = printfn "Connected!"
```